modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/HeadersSideBar.scala

```diff
@@ -218,39 +225,36 @@ object HeadersSideBar {
   }
 
   private val component = ScalaComponent
-    .builder[HeadersSideBar]("HeadersSideBar")
-    .initialState(State(None, None))
+    .builder[HeadersSideBar]
+    .getDerivedStateFromPropsAndState[State]{ (p, sOpt) =>
+      val operator = p.model.operator
+      val observer = 
+        p.selectedObserver match {
+          case Right(Right(a)) => a.observer
+          case Right(Left(a))  => a.observer
+          case Left(o)         => o.some
+        }
+
+      sOpt.fold(State(operator, observer)){ s =>
```

Comment:
When is `sOpt` None?

Reply:
When initializing `State` for the first time. `getDerivedStateFromPropsAndState` combines `initialState` and `getDerivedStateFromProps`.